ocaml/libs/tracing/tracing.ml

```diff
@@ -15,6 +15,63 @@ module D = Debug.Make (struct let name = "tracing" end)
 
 open D
 
+module W3CBaggage = struct
+  let explode str = String.fold_right (fun c acc -> c :: acc) str []
+
+  module Key = struct
+    let is_valid_key str =
+      let is_tchar = function
+        | '0' .. '9'
+        | 'a' .. 'z'
+        | 'A' .. 'Z'
+        | '!'
+        | '#'
+        | '$'
+        | '%'
+        | '&'
+        | '\''
+        | '*'
+        | '+'
+        | '-'
+        | '.'
+        | '^'
+        | '_'
+        | '`'
+        | '|'
+        | '~' ->
+            true
+        | _ ->
+            false
+      in
+      let chars = explode str in
+      List.for_all (fun c -> is_tchar c) chars
+  end
+
+  module Value = struct
+    type t = string
+
+    let serialise str =
```

Comment:
make instead of serialize? https://github.com/lindig/ocaml-style#introduce-and-document-interfaces